tests/ignite/metrics/test_multilabel_confusion_matrix.py

```diff
@@ -0,0 +1,438 @@
+
+        # Another test on batch of 2 images
+        num_classes = 3
+        cm = MultiLabelConfusionMatrix(num_classes=num_classes, device=metric_device)
+
+        # Create a batch of two images:
+        th_y_true1 = torch.tensor(y_true)
+        th_y_true2 = torch.tensor(y_true.transpose(0, 1, 3, 2))
+        th_y_true = torch.cat([th_y_true1, th_y_true2], dim=0)
+        th_y_true = th_y_true.to(device)
+
+        th_y_pred1 = torch.tensor(y_pred)
+        th_y_pred2 = torch.tensor(y_pred.transpose(0, 1, 3, 2))
+        th_y_pred = torch.cat([th_y_pred1, th_y_pred2], dim=0)
+        th_y_pred = th_y_pred.to(device)
+
+        # Update metric & compute
+        output = (th_y_pred, th_y_true)
+        cm.update(output)
+        ignite_CM = cm.compute().cpu().numpy()
+
+        # Compute confusion matrix with sklearn
+        th_y_true = idist.all_gather(th_y_true)
+        th_y_pred = idist.all_gather(th_y_pred)
+
+        np_y_true = th_y_true.cpu().numpy().transpose((0, 2, 3, 1)).reshape(-1, 3)
+        np_y_pred = th_y_pred.cpu().numpy().transpose((0, 2, 3, 1)).reshape(-1, 3)
+        sklearn_CM = multilabel_confusion_matrix(np_y_true, np_y_pred)
+
+        assert np.all(ignite_CM == sklearn_CM)
+
+    _test("cpu")
+    if device.type != "xla":
+        _test(idist.device())
+
+
+def _test_distrib_accumulator_device(device):
+
+    metric_devices = [torch.device("cpu")]
+    if device.type != "xla":
+        metric_devices.append(idist.device())
+    for metric_device in metric_devices:
+
+        cm = MultiLabelConfusionMatrix(num_classes=3, device=metric_device)
+        assert cm._device == metric_device
+        assert (
+            cm.confusion_matrix.device == metric_device
+        ), f"{type(cm.confusion_matrix.device)}:{cm._num_correct.device} vs {type(metric_device)}:{metric_device}"
+
+        y_true, y_pred = get_y_true_y_pred()
+        cm.update((torch.tensor(y_pred), torch.tensor(y_true)))
+
+        assert (
+            cm.confusion_matrix.device == metric_device
+        ), f"{type(cm.confusion_matrix.device)}:{cm._num_correct.device} vs {type(metric_device)}:{metric_device}"
+
+
+def test_simple_2D_input():
+
+    num_iters = 100
+    num_samples = 100
+    num_classes = 10
+    for _ in range(num_iters):
+        target = np.random.randint(0, 2, size=(num_samples, num_classes))
+        prediction = np.random.randint(0, 2, size=(num_samples, num_classes))
+        sklearn_CM = multilabel_confusion_matrix(target, prediction)
+        mlcm = MultiLabelConfusionMatrix(num_classes, normalized=False)
+        mlcm.update([torch.tensor(prediction), torch.tensor(target)])
+        ignite_CM = mlcm.compute().numpy()
+        assert np.all(sklearn_CM.astype(np.int64) == ignite_CM.astype(np.int64))
+
+    return
+
+
+def test_simple_ND_input():
+
+    # 3D tests
+    num_iters = 100
+    num_samples = 100
+    num_classes = 10
+    size_3d = 4
+
+    for _ in range(num_iters):  # 3D tests
```

Comment:
Why we would like to run 100 (num_iters) the same (random data) test ? 2-3 times may be enough ?

Reply:
I thought the more number of times we run the better we are off since we might be covering edge cases there. Btw, I am not setting the numpy random number generator seed here, so each run should produce different data.

Reply:
Well, covering edge cases in practice should be done explicitly than using a random input. That's a good point about numpy input, let's generate data with torch as we do it in other tests and we also set seed here for torch generator.

Reply:
Changed the matrix generation part from numpy to torch.